specs/Popup-AdditionalLayoutProperties-Spec.md

```diff
@@ -0,0 +1,226 @@
+}
+
+void UpdateVisualState(bool useTransitions)
+{
+    if (m_overflowPopup.ActualPlacement == PopupPlacementMode.Top)
+    {
+        VisualStateManager.GoToState(this, "ExpandedUp", useTransitions);
+    }
+    else
+    {
+        VisualStateManager.GoToState(this, "ExpandedDown", useTransitions);
+    }
+}
+```
+
+## Popup.PlacementTarget property
+
+Use this property to describe which element the `Popup` should be positioned relative to.
+Defaults to `null`.
+
+If this is `null`, then `DesiredPlacement` is ignored, `ActualPlacement` is always `None`, and
+`PlacementChanged` is never raised.  If the `Popup` is in the visual tree, `PlacementTarget` will override what its
+position would otherwise be set to by layout.  Setting `PlacementTarget` to an element under a different XAML root than
+`Popup.XamlRoot` is invalid and will throw an `ArgumentException`.
+
+Ignored if `DesiredPlacement` is `None`.
+
+_Spec note: this property is analogous to the
+[FlyoutBase.Target](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.Primitives.FlyoutBase.Target)
+property, but `Popup.Target` looked confusing, so we added the `Placement` prefix._
+
+## Popup.DesiredPlacement property
+
+Use this property to describe how you would ideally like the `Popup`
+positioned relative to `PlacementTarget`.  Defaults to `None`.
+
+If this is `None`, then `PlacementTarget` is ignored,
+`ActualPlacement` is always `None` and `PlacementChanged` is never raised. 
+If both `DesiredPlacement` and `PlacementTarget` are set and `HorizontalOffset` and/or `VerticalOffset`
+are also set, then the latter two properties will offset the `Popup` from where it would have been
+placed by `DesiredPlacement` and `PlacementTarget` alone.
+
+Ignored if `PlacementTarget` is null.
+
+## Popup.ActualPlacement property
+
+Use this read-only property to determine where the popup was positioned.
+Will always be `None` if either `PlacementTarget` and `DesiredPlacement` are not set.
+
+## Popup.PlacementChanged event
+
+Raised whenever XAML changes the value of `ActualPlacement`,
+which allows apps to respond to where a `Popup` was placed.
+
+For example, use this to determine the visual state to go into, 
+based on whether a `Popup` is appearing above or below `PlacementTarget`.
+
+This event is raised before the screen is refreshed, meaning that any visual changes made
+in response to this event can be made before anything is drawn to the screen at the new position.
+Will never be raised if either `PlacementTarget` and `DesiredPlacement` are not set.
+
+## PopupPlacementMode enum
+
+_Spec note: This is designed to align with the existing
+[FlyoutPlacementMode](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.Primitives.FlyoutPlacementMode)
+enum, with the exception of the absence of "Full".  "Full" is absent since developers should use a Flyout
+if they want something full-screen._
+
+```csharp
+enum PopupPlacementMode
+{
+    None,
+    Top,
```

Comment:
Does `Top`, `Left`, `Bottom`, `Right`  mean that the popup will be centered?

Reply:
Yes, it has the same semantics as [FlyoutPlacementMode](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.primitives.flyoutplacementmode?view=winrt-19041): `Top` for example means that it's horizontally centered above the `PlacementTarget`.

Reply:
Thank you for your response, it wasn't clear to me, since the documentation did not state it explicitly. This was always a downside in WPF, where you could not explicitly state the alignment of the popup.